test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -4847,6 +4847,203 @@ TEST_F(AtenXlaTensorTest, TestScatterAddInPlace) {
   }
 }
 
+TEST_F(AtenXlaTensorTest, TestScatterReduceSum) {
+  torch::Tensor a = torch::rand({3, 5}, torch::TensorOptions(torch::kFloat));
+  torch::Tensor b = torch::rand({3, 5}, torch::TensorOptions(torch::kFloat));
+  torch::Tensor c = torch::empty({3, 5}, torch::TensorOptions(torch::kLong));
+  for (int dim = 0; dim < 2; ++dim) {
+    for (int i = 0; i < 3; i++) {
+      for (int j = 0; j < 5; j++) {
+        c[i][j] = (i + j) % c.sizes()[dim];
```

Comment:
why don't we just use `torch::scatter_reduce` here as well? This is safer in case upstream `scatter_reduce` implementation changed.

Reply:
we are using `torch::scatter_reduce`, see line 4860. Here we are initializing the input variable to make sure the values are valid.

Reply:
Ah ok